src/Plutus/Types/Credential.purs

```diff
@@ -4,15 +4,31 @@ module Plutus.Types.Credential
   ) where
 
 import Prelude
-import Data.Maybe (Maybe(Nothing))
+
+import Aeson (class DecodeAeson, class EncodeAeson, encodeAeson')
+import Aeson.Decode ((</$\>), (</*\>))
+import Aeson.Encode ((>/\<))
+import Control.Lazy (defer)
 import Data.Generic.Rep (class Generic)
 import Data.Show.Generic (genericShow)
-import Types.Scripts (ValidatorHash)
-import Types.PlutusData (PlutusData(Constr))
+import Data.Tuple.Nested ((/\))
+import FromData (class FromData, genericFromData)
+import Plutus.Types.DataSchema
+  ( class HasPlutusSchema
+  , type (:+)
+  , type (:=)
+  , type (@@)
+  , I
+  , PNil
+  )
+import ToData (class ToData, genericToData)
+import TypeLevel.Nat (S, Z)
+import Aeson.Decode as D
```

Comment:
Since we had a bunch of Aeson instances we had to code in, we just c/p-d purs-bridged code here. That's what got generated and I agree it's not pretty. We should however make a plan on how purs-bridge (ctl-bridge) can generate all of that stuff as it's also hooked with a bunch of good tests.

Imo the less code we have to manually write the better.